File: examples/chat_completion.py
"""
Brainiall LLM Gateway — Python Chat Completion Examples

Base URL: https://apim-ai-apis.azure-api.net/v1
Get your API key at https://brainiall.com
"""

from openai import OpenAI

# Initialize client
client = OpenAI(
    base_url="https://apim-ai-apis.azure-api.net/v1",
    api_key="YOUR_KEY"  # Replace with your API key
)


def basic_chat():
    """Simple chat completion."""
    response = client.chat.completions.create(
        model="claude-sonnet-4-6",
        messages=[{"role": "user", "content": "What is the capital of Japan?"}]
    )
    print(response.choices[0].message.content)
    print(f"Tokens: {response.usage.total_tokens}")


def streaming_chat():
    """Streaming chat completion with real-time output."""
    stream = client.chat.completions.create(
        model="claude-sonnet-4-6",
        messages=[
            {"role": "system", "content": "You are a creative writer."},
            {"role": "user", "content": "Write a short poem about the ocean."}
        ],
        stream=True,
        max_tokens=256
    )

    for chunk in stream:
        if chunk.choices[0].delta.content:
            print(chunk.choices[0].delta.content, end="", flush=True)
    print()


def multi_turn_conversation():
    """Multi-turn conversation with message history."""
    messages = [
        {"role": "system", "content": "You are a math tutor. Be concise."},
        {"role": "user", "content": "What is the derivative of x^3?"}
    ]

    # First turn
    response = client.chat.completions.create(
        model="claude-haiku-4-5",
        messages=messages
    )
    assistant_msg = response.choices[0].message.content
    print(f"Assistant: {assistant_msg}")

    # Second turn
    messages.append({"role": "assistant", "content": assistant_msg})
    messages.append({"role": "user", "content": "What about x^4 + 2x^2?"})

    response = client.chat.completions.create(
        model="claude-haiku-4-5",
        messages=messages
    )
    print(f"Assistant: {response.choices[0].message.content}")


def json_mode():
    """Force JSON output from the model."""
    response = client.chat.completions.create(
        model="claude-sonnet-4-6",
        messages=[
            {"role": "system", "content": "Extract structured data. Return valid JSON."},
            {"role": "user", "content": "John Smith, 35 years old, works at Google in Mountain View as a Senior Engineer."}
        ],
        response_format={"type": "json_object"},
        max_tokens=256
    )

    import json
    data = json.loads(response.choices[0].message.content)
    print(json.dumps(data, indent=2))


def compare_models():
    """Compare responses from different models."""
    models = ["claude-haiku-4-5", "nova-micro", "deepseek-v3"]
    prompt = "In one sentence, what is machine learning?"

    for model in models:
        response = client.chat.completions.create(
            model=model,
            messages=[{"role": "user", "content": prompt}],
            max_tokens=100
        )
        print(f"[{model}] {response.choices[0].message.content}")
        print(f"  Tokens: {response.usage.total_tokens}")
        print()


def vision_example():
    """Analyze an image using a vision-capable model."""
    response = client.chat.completions.create(
        model="claude-sonnet-4-6",
        messages=[
            {
                "role": "user",
                "content": [
                    {"type": "text", "text": "Describe this image briefly."},
                    {
                        "type": "image_url",
                        "image_url": {
                            "url": "https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/Camponotus_flavomarginatus_ant.jpg/320px-Camponotus_flavomarginatus_ant.jpg"
                        }
                    }
                ]
            }
        ],
        max_tokens=256
    )
    print(response.choices[0].message.content)


def list_models():
    """List all available models."""
    models = client.models.list()
    for model in sorted(models.data, key=lambda m: m.id):
        print(f"{model.id:40s} — {model.owned_by}")


if __name__ == "__main__":
    print("=== Basic Chat ===")
    basic_chat()

    print("\n=== Streaming ===")
    streaming_chat()

    print("\n=== Multi-Turn ===")
    multi_turn_conversation()

    print("\n=== JSON Mode ===")
    json_mode()

    print("\n=== Compare Models ===")
    compare_models()

    print("\n=== Vision ===")
    vision_example()

    print("\n=== List Models ===")
    list_models()

File: examples/chat_completion.js
/**
 * Brainiall LLM Gateway — JavaScript Chat Completion Examples
 *
 * Base URL: https://apim-ai-apis.azure-api.net/v1
 * Get your API key at https://brainiall.com
 *
 * Install: npm install openai
 */

import OpenAI from "openai";

const client = new OpenAI({
  baseURL: "https://apim-ai-apis.azure-api.net/v1",
  apiKey: "YOUR_KEY", // Replace with your API key
});

// Basic chat completion
async function basicChat() {
  const response = await client.chat.completions.create({
    model: "claude-sonnet-4-6",
    messages: [{ role: "user", content: "What is the capital of Japan?" }],
  });
  console.log(response.choices[0].message.content);
  console.log(`Tokens: ${response.usage.total_tokens}`);
}

// Streaming chat completion
async function streamingChat() {
  const stream = await client.chat.completions.create({
    model: "claude-sonnet-4-6",
    messages: [
      { role: "system", content: "You are a creative writer." },
      { role: "user", content: "Write a haiku about programming." },
    ],
    stream: true,
    max_tokens: 256,
  });

  for await (const chunk of stream) {
    const content = chunk.choices[0]?.delta?.content;
    if (content) process.stdout.write(content);
  }
  console.log();
}

// Tool/function calling
async function toolCalling() {
  const tools = [
    {
      type: "function",
      function: {
        name: "get_weather",
        description: "Get the current weather for a location",
        parameters: {
          type: "object",
          properties: {
            location: {
              type: "string",
              description: "City name, e.g. 'San Francisco, CA'",
            },
            unit: {
              type: "string",
              enum: ["celsius", "fahrenheit"],
            },
          },
          required: ["location"],
        },
      },
    },
  ];

  const response = await client.chat.completions.create({
    model: "claude-sonnet-4-6",
    messages: [{ role: "user", content: "What's the weather in Tokyo?" }],
    tools,
    tool_choice: "auto",
  });

  const message = response.choices[0].message;
  if (message.tool_calls) {
    for (const toolCall of message.tool_calls) {
      console.log(`Function: ${toolCall.function.name}`);
      console.log(`Arguments: ${toolCall.function.arguments}`);
    }
  }
}

// JSON mode
async function jsonMode() {
  const response = await client.chat.completions.create({
    model: "claude-sonnet-4-6",
    messages: [
      {
        role: "system",
        content: "Extract structured data. Return valid JSON.",
      },
      {
        role: "user",
        content:
          "John Smith, 35 years old, works at Google in Mountain View as a Senior Engineer.",
      },
    ],
    response_format: { type: "json_object" },
  });

  const data = JSON.parse(response.choices[0].message.content);
  console.log(JSON.stringify(data, null, 2));
}

// Multi-turn conversation
async function multiTurn() {
  const messages = [
    { role: "system", content: "You are a helpful math tutor." },
    { role: "user", content: "What is the derivative of x^3?" },
  ];

  let response = await client.chat.completions.create({
    model: "claude-haiku-4-5",
    messages,
  });

  const firstAnswer = response.choices[0].message.content;
  console.log(`Turn 1: ${firstAnswer}`);

  messages.push({ role: "assistant", content: firstAnswer });
  messages.push({ role: "user", content: "And the integral of that?" });

  response = await client.chat.completions.create({
    model: "claude-haiku-4-5",
    messages,
  });
  console.log(`Turn 2: ${response.choices[0].message.content}`);
}

// List all available models
async function listModels() {
  const models = await client.models.list();
  for (const model of models.data.sort((a, b) => a.id.localeCompare(b.id))) {
    console.log(`${model.id.padEnd(40)} — ${model.owned_by}`);
  }
}

// Run examples
console.log("=== Basic Chat ===");
await basicChat();

console.log("\n=== Streaming ===");
await streamingChat();

console.log("\n=== Tool Calling ===");
await toolCalling();

console.log("\n=== JSON Mode ===");
await jsonMode();

console.log("\n=== Multi-Turn ===");
await multiTurn();

console.log("\n=== List Models ===");
await listModels();
